src/t_hash.c

```diff
@@ -2165,8 +2312,17 @@ void hmgetCommand(client *c) {
     if (checkType(c,o,OBJ_HASH)) return;
 
     addReplyArrayLen(c, c->argc-2);
-    for (i = 2; i < c->argc; i++) {
-        addHashFieldToReply(c, o, c->argv[i]->ptr);
+    for (i = 2; i < c->argc ; i++) {
+
+        res = addHashFieldToReply(c, o, c->argv[i]->ptr);
+
+        /* If hash got lazy expired since all fields are expired (o is invalid),
+         * then fill the rest with trivial nulls and return */
+        if (res == GETF_EXPIRED_HASH) {
+            for ( ++i ; i < c->argc; i++)
+                addReply(c, shared.null[c->resp]);
+            return;
```

Comment:
```suggestion
    for (i = 2; i < c->argc; i++) {
        res = addHashFieldToReply(c, o, c->argv[i]->ptr);

        /* If hash got lazy expired since all fields are expired (o is invalid),
         * then fill the rest with trivial nulls and return */
        if (res == GETF_EXPIRED_HASH) { 
            while (++i < c->argc)
                addReplyNull(c);
            return;
```